staging/src/k8s.io/legacy-cloud-providers/openstack/openstack.go

```diff
@@ -290,11 +290,11 @@ func (os *OpenStack) setConfigFromSecret() error {
 
 		err = gcfg.ReadStringInto(cfg, string(content))
 		if err != nil {
-			return fmt.Errorf("cannot parse data from the secret")
+			return fmt.Errorf("cannot parse data from the secret: %s", err)
```

Comment:
Isn't it a bit dangerous to print the error like that? This could lead to secret leak in the logs. I'd personally prefer we leave this line as it.

Reply:
I just checked, it's pretty safe to show the error. First of all, the error is shown only if the format is broken, which is hard to achieve since the file is automatically generated. Second, if the format is broken the error output looks like `cannot parse data from the secret: 8:11: expected '='`, so it doesn't print the content. Third, the `password` has the `datapolicy` tag https://github.com/openshift/kubernetes/blob/master/staging/src/k8s.io/legacy-cloud-providers/openstack/openstack.go#L167, which prevents it to be shown in the logs anyway: https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1753-logs-sanitization#datapolicy-verification-library